libs/clients/smartsolutions/src/lib/smartSolutions.api.ts

```diff
@@ -210,49 +210,92 @@ export class SmartSolutionsApi {
       data: undefined,
     }
   }
-
-  async verifyPkPass(
-    payload: DynamicBarcodeDataInput,
-  ): Promise<Result<VerifyPassData>> {
+  /**
+   *
+   * @param payload What the created/updated pass should contain
+   * @returns The created/updated pass
+   */
+  private async postPass(payload: PassDataInput): Promise<Result<Pass>> {
     const graphql = JSON.stringify({
-      query: VERIFY_PKPASS,
+      query: UPSERT_PASS,
       variables: {
-        dynamicBarcodeData: payload,
+        inputData: {
+          passTemplateId: this.config.passTemplateId,
+          ...payload,
+        },
       },
     })
 
-    const res = await this.query<VerifyPassResponseData>(graphql)
+    const res = await this.query<UpsertPassResponseData>(graphql)
 
     if (res.ok) {
       return {
         ok: true,
-        data: {
-          valid: true,
-          pass: res.data.updateStatusOnPassWithDynamicBarcode,
-        },
+        data: res.data.upsertPass,
       }
     }
 
     return res
   }
 
-  async upsertPkPass(payload: PassDataInput): Promise<Result<Pass>> {
+  private async updateExistingPass(
```

Comment:
Shouldn't this also update the `expiryDate` in Smart?

Reply:
it does, the expiryDate is part of the payload which the client sends over
<img width="680" alt="image" src="https://user-images.githubusercontent.com/9449413/225742535-7ccc358b-200a-411d-a698-30ff7d7ce16f.png">

Reply:
We just have to be careful to do the same with other clients, which is on the to-do list